mslib/msui/editor.py

```diff
@@ -329,6 +402,79 @@ def tree_selection_changed(self, selected, deselected):
         self.moveUpTb.setEnabled(move)
         self.moveDownTb.setEnabled(move)
 
+    def update_view(self):
+        source_model = self.json_model
+        data = source_model.serialize()
+        parent = QtCore.QModelIndex()
+        for r in range(source_model.rowCount(parent)):
+            root_index = source_model.index(r, 0, parent)
+            root_item = source_model.itemFromIndex(root_index)
+
+            if root_item.data(InvalidityRole) is None:
+                root_item.setData([False, False, False], InvalidityRole)
+            if root_item.data(DummyRole) is None:
+                root_item.setData(False, DummyRole)
+
+            color = QtCore.Qt.transparent
+            key = root_index.data()
+            if key in dict_option_structure:
+                # check for empty values
+                empty = False
+                child_keys = set()
+                rows = source_model.rowCount(root_index)
+                for row in range(rows):
+                    child_key_data = source_model.index(row, 0, root_index).data()
+                    child_keys.add(child_key_data)
+                    if child_key_data == "":
+                        empty = True
+
+                # check for dummy values
+                dummy = False
+                default = dict_option_structure[key]
+                values_dict = data[key]
+                for value in values_dict:
+                    if value in default:
+                        if default[value] == values_dict[value]:
+                            dummy = True
+                            color = QtCore.Qt.gray
+                            break
+
+                # condition for checking duplicate and empty
+                duplicate = False
+                if len(child_keys) != rows or empty:
+                    duplicate = True
+                    color = QtCore.Qt.red
+                root_item.setData([empty, duplicate, False], InvalidityRole)
+                root_item.setData(dummy, DummyRole)
+            elif key in list_option_structure:
+                empty, duplicate, dummy = [False] * 3
+                values_list = data[key]
+                if any([values_list[i] == list_option_structure[key][0] for i in range(len(values_list))]):
```

Comment:
`if any([v == list_option_structure[key][0] for v in values_list]):` ?

Reply:
Better. Thanks.